lib/ansible/plugins/connection/ssh.py

```diff
@@ -475,12 +475,17 @@ def _add_args(self, b_command, b_args, explanation):
         display.vvvvv(u'SSH: %s: (%s)' % (explanation, ')('.join(to_text(a) for a in b_args)), host=self._play_context.remote_addr)
         b_command += b_args
 
-    def _build_command(self, binary, *other_args):
+    def _build_command(self, binary, *other_args, **kwargs):
```

Comment:
From what I understand, you cannot use named parameters with varargs in Python 2, so does that mean you'd prefer to use a positional parameter (i.e. `_build_command(self, binary, command, *other_args)`)?

Reply:
The better API is to take a list for other_args and a dict for kwargs so that we never have conflicts between the two but I see why you went the **kwargs route (since other_args was already using the unpakcing idiom).

This is an internal function so we have the freedom to change the function's signature  At first glance there's a bunch of options:
``` python
def _build_command(self, binary, command, *other_args):
    [...]

def _build_command(self, binary, command, other_args=None):
    if other_args is None:
        other_args = []

def _build_command(self, binary, command='ssh', other_args=None):
    if other_args is None:
        other_args = []
```

Looking at what the code does, though, I think this would be even better:

``` python
def _build_command(self, command, other_args=None):
    if other_args is None:
        other_args = []
    if command == 'ssh':
        binary = self.play_context.ssh_executable
    else:
        binary = self.get_option('{0}_executable'.format(command))
```
Here's my thinking:
* If we're limiting the possible commands to the set of ssh, scp, and sftp, then the correct command is really the primary piece of information that this function needs.  Therefore, command should both be mandatory and the first argument.
* Since command is limited to a few specific values, we can choose the proper value of binary based on the value of command.  Therefore we don't need to pass in binary; we can set it inside of this function.  My glance at all the callers of _build_command(), this looks like it's the correct way to default binary for all three of the possible commands but you'll probably want to hceck my eyesight ;-)
* We need to change the signature of the function and then update the callers so we might as well correct all the deficiencies in the function's signature. (Hence, changing other_args into a kwarg taking a list of arguments).

Hope that makes sense.

Reply:
That works for me. @bcoca?

Reply:
Also agree, I asked @abadger to look over this after your question as he is much better at 'idomaticisims' than I am.

Reply:
Okay, it's been refactored as discussed. Waiting to make sure the CI passes...